### PR TITLE
#5780 Remove service_worker for firefox's manifest.json build process

### LINKED ISSUE #5780: Firefox manifest.json includes background.service_worker as a manifest version 2
In Firefox build, the service_worker property is not recognized as a valid manifest property for Manifest V2. As a result, a following warning was shown.

image

### PR DESCRIPTION
**Pull request by martgil**
This PR removes `background.service_worker` for Firefox's manifest.json build process to remove such warning for unknown property `service_worker` for Manifest V2.

close #5780

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
@sosnovsky Ready for review. thank you!

**Comment by martgil** [comment]
For testing, just load a debug Firefox build from this PR and notice the error message reported in #5780 will no longer occur.

File: tooling/build-types-and-manifests.ts [hunk 1]
```diff
@@ -44,8 +44,9 @@ addManifest('firefox-consumer', manifest => {
       strict_min_version: '112.0', // eslint-disable-line @typescript-eslint/naming-convention
     },
   };
+  delete manifest.background;
   manifest.background = {
-    ...manifest.background,
+    type: 'module',
     scripts: ['/js/service_worker/background.js'],
   };
```
Comment (sosnovsky) [inline tooling/build-types-and-manifests.ts:51]: no need to delete `manifest.background`, as it's reassigned on the next line, so `delete manifest.background;` can be removed

Comment (martgil) [reply tooling/build-types-and-manifests.ts]: Oh - okay, I'll remove it now. thanks!

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**